apps/avifdec.c

```diff
@@ -178,17 +189,31 @@ int main(int argc, char * argv[])
                    decoder->durationInTimescales,
                    decoder->imageCount,
                    (decoder->imageCount == 1) ? "" : "s");
-            printf(" * Frames:\n");
+            if (decoder->imageCount > 1) {
+                printf(" * %s Frames: (%u expected frames)\n",
+                       (decoder->progressiveState != AVIF_PROGRESSIVE_STATE_UNAVAILABLE) ? "Progressive Image" : "Image Sequence",
+                       decoder->imageCount);
+            } else {
+                printf(" * Frame:\n");
+            }
 
-            int frameIndex = 0;
-            while (avifDecoderNextImage(decoder) == AVIF_RESULT_OK) {
-                printf("   * Decoded frame [%d] [pts %2.2f (%" PRIu64 " timescales)] [duration %2.2f (%" PRIu64 " timescales)]\n",
-                       frameIndex,
+            int currIndex = 0;
+            avifResult nextImageResult = avifDecoderNextImage(decoder);
+            while (nextImageResult == AVIF_RESULT_OK) {
+                printf("   * Decoded frame [%d] [pts %2.2f (%" PRIu64 " timescales)] [duration %2.2f (%" PRIu64 " timescales)] [%ux%u]\n",
+                       currIndex,
                        decoder->imageTiming.pts,
                        decoder->imageTiming.ptsInTimescales,
                        decoder->imageTiming.duration,
-                       decoder->imageTiming.durationInTimescales);
-                ++frameIndex;
+                       decoder->imageTiming.durationInTimescales,
+                       decoder->image->width,
+                       decoder->image->height);
+                ++currIndex;
+                nextImageResult = avifDecoderNextImage(decoder);
+            }
```

Comment:
Lines 201-213 can be simplified slightly (only one avifDecoderNextImage call) if you are willing to do an assignment in the while loop's conditional expression:

```
            avifResult nextImageResult;
            while ((nextImageResult = avifDecoderNextImage(decoder)) == AVIF_RESULT_OK) {
                printf("   * Decoded frame [%d] [pts %2.2f (%" PRIu64 " timescales)] [duration %2.2f (%" PRIu64 " timescales)] [%ux%u]\n",
                       currIndex,
                       decoder->imageTiming.pts,
                       decoder->imageTiming.ptsInTimescales,
                       decoder->imageTiming.duration,
                       decoder->imageTiming.durationInTimescales,
                       decoder->image->width,
                       decoder->image->height);
                ++currIndex;
            }
```